handlers.go

```diff
@@ -58,21 +58,32 @@ func BeginAuth(w http.ResponseWriter, r *http.Request) {
     }
 }
 
-func getSingleSecQues(roll string) string {
+func getSingleSecQues(roll string,c chan string) {
+    defer func() {
+        if (recover() != nil) {
+            log.Printf("Exiting gracefully\n")   // in case panic occurs while writing to a closed channel
```

Comment:
This block of code is unintuitive. I would rather wrap the `c <- string(body)` part with this check.